week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,347 @@
+
+
+#MODEL
+model <- lm(num_trips ~ poly(tmin, 5, raw = T), data = trips_per_day_train)
+
+trips_per_day_train <- trips_per_day_train %>%
+  add_predictions(model) %>%
+  mutate(split = "train")
+trips_per_day_validate <- trips_per_day_validate %>%
+  add_predictions(model) %>%
+  mutate(split = "validate")
+plot_data <- bind_rows(trips_per_day_train, trips_per_day_validate)
+
+ggplot(plot_data, aes(x = tmin, y = num_trips)) +
+  geom_point(aes(color = split)) +
+  geom_line(aes(y = pred)) +
+  xlab('Minimum temperature') +
+  ylab('Daily trips') +
+  scale_y_continuous()
+```
+
+```{r}
+# set.seed(42)
+# 
+# num_days <- nrow(trips_per_day_trainval)
+# frac_train <- 0.8
+# num_train <- floor(num_days * frac_train)
+# 
+# # randomly sample rows for the training set
+# ndx <- sample(1:num_days, num_train, replace=F)
+# 
+# # used to fit the model
+# trips_per_day_train <- trips_per_day_trainval[ndx, ]
+# 
+# # used to evaluate the fit
+# trips_per_day_validate <- trips_per_day_trainval[-ndx, ]
+# 
+# 
+# 
+# # fit a model for each polynomial degree
+# K <- 1:8
+# train_err <- c()
+# validate_err <- c()
+# for (k in K) {
+# 
+#     # fit on the training data
+#     model <- lm(num_trips ~ poly(tmin, k, raw = T), data=trips_per_day_train)
+# 
+#     # evaluate on the training data
+#     train_err[k] <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+# 
+#     # evaluate on the validate data
+#     validate_err[k] <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+# }
+# 
+# #ERRORS
+# plot_data <- data.frame(K, train_err, validate_err) %>%
+#   gather("split", "error", -K)
+# 
+# ggplot(plot_data, aes(x=K, y=error, color=split)) +
+#   geom_line() +
+#   scale_x_continuous(breaks=K) +
+#   xlab('Polynomial Degree') +
+#   ylab('RMSE')
+# 
+# 
+# #MODEL
+# model <- lm(num_trips ~ poly(tmin, 5, raw = T), data = trips_per_day_train)
+# 
+# trips_per_day_train <- trips_per_day_train %>%
+#   add_predictions(model) %>%
+#   mutate(split = "train")
+# trips_per_day_validate <- trips_per_day_validate %>%
+#   add_predictions(model) %>%
+#   mutate(split = "validate")
+# plot_data <- bind_rows(trips_per_day_train, trips_per_day_validate)
+# 
+# ggplot(plot_data, aes(x = tmin, y = num_trips)) +
+#   geom_point(aes(color = split)) +
+#   geom_line(aes(y = pred)) +
+#   xlab('Minimum temperature') +
+#   ylab('Daily trips') +
+#   scale_y_continuous()
+
+```
+
+
+
+At this point we found that the best polynomial degree, which has the lowest RMSE on validation data, is 8.
+
+Build different models creatively:
+1. Weekdays vs weekends and holidays 
+a. Join holidays table and trips_per_day_trainval. If that day is a holiday or weekend day, boolean = 0 in col holiday.
+```{r}
+library(lubridate)
+holidays <- read.table(file = "holidays.csv", sep = ',', header = FALSE, col.names = c("row", "ymd", "holiday_name")) %>% select(ymd) %>% mutate(ymd = as.Date(ymd), is_holiday = "Holiday")

```

Comment:
You should include a link to the holidays data or include it in your repo. If for whatever reason you can't, you should knit the Rmd file and include the output. Reproducibility is a key part of data science.